[CI] SharedClusterSnapshotRestoreIT.testSnapshotClosedIndex failures

Console log: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.3+multijob+fast+part1/150/console

Build scans: https://gradle.com/s/qyqk2hevjfdxc

Failure:
```
14:21:46   2> REPRODUCE WITH: ./gradlew :server:integTest --tests "org.elasticsearch.snapshots.SharedClusterSnapshotRestoreIT.testSnapshotClosedIndex" -Dtests.seed=90EF99787317695B -Dtests.security.manager=true -Dtests.locale=es-US -Dtests.timezone=Asia/Aqtau -Dcompiler.java=12 -Druntime.java=8
14:21:46   2> java.lang.AssertionError: 
14:21:46     Expected: <0>
14:21:46          but: was <2>
14:21:46         at __randomizedtesting.SeedInfo.seed([90EF99787317695B:ED716DF18291CA3B]:0)
14:21:46         at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
14:21:46         at org.junit.Assert.assertThat(Assert.java:956)
14:21:46         at org.junit.Assert.assertThat(Assert.java:923)
14:21:46         at org.elasticsearch.snapshots.SharedClusterSnapshotRestoreIT.testSnapshotClosedIndex(SharedClusterSnapshotRestoreIT.java:1581)
```

Reproduction lines
```
REPRODUCE WITH: ./gradlew :server:integTest --tests "org.elasticsearch.snapshots.SharedClusterSnapshotRestoreIT.testSnapshotClosedIndex" \
  -Dtests.seed=90EF99787317695B \
  -Dtests.security.manager=true \
  -Dtests.locale=es-US \
  -Dtests.timezone=Asia/Aqtau \
  -Dcompiler.java=12 \
  -Druntime.java=8
```

```
REPRODUCE WITH: ./gradlew :server:integTest --tests "org.elasticsearch.snapshots.SharedClusterSnapshotRestoreIT.testSnapshotClosedIndex" \
  -Dtests.seed=90EF99787317695B \
  -Dtests.security.manager=true \
  -Dtests.locale=es-US \
  -Dtests.timezone=Asia/Aqtau \
  -Dcompiler.java=12 \
  -Druntime.java=8
```

This have failed infrequently ( 5 times in July ) : https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-30d,mode:quick,to:now))&_a=(columns:!(_source),index:e58bf320-7efd-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:testSnapshotClosedIndex),sort:!(time,desc))

--- a/server/src/test/java/org/elasticsearch/snapshots/SharedClusterSnapshotRestoreIT.java
+++ b/server/src/test/java/org/elasticsearch/snapshots/SharedClusterSnapshotRestoreIT.java
@@ -1569,7 +1569,7 @@ public class SharedClusterSnapshotRestoreIT extends AbstractSnapshotIntegTestCas
         createIndex("test-idx", "test-idx-closed");
         ensureGreen();
         logger.info("-->  closing index test-idx-closed");
-        assertAcked(client.admin().indices().prepareClose("test-idx-closed"));
+        assertAcked(client.admin().indices().prepareClose("test-idx-closed").setWaitForActiveShards(ActiveShardCount.ALL));
         ClusterStateResponse stateResponse = client.admin().cluster().prepareState().get();
         assertThat(stateResponse.getState().metaData().index("test-idx-closed").getState(), equalTo(IndexMetaData.State.CLOSE));
         assertThat(stateResponse.getState().routingTable().index("test-idx-closed"), notNullValue());

